lib/TextProcessing/FreePromptProvider.php

```diff
@@ -27,14 +29,17 @@ public function getName(): string {
 
 	public function process(string $prompt): string {
 		$adminModel = $this->config->getAppValue(Application::APP_ID, 'default_completion_model_id', Application::DEFAULT_COMPLETION_MODEL_ID) ?: Application::DEFAULT_COMPLETION_MODEL_ID;
-		$completion = $this->openAiAPIService->createChatCompletion(null, $prompt, 1, $adminModel, 100, false);
-		if (isset($completion['choices']) && is_array($completion['choices']) && count($completion['choices']) > 0) {
-			$choice = $completion['choices'][0];
-			if (isset($choice['message'], $choice['message']['content'])) {
-				return $choice['message']['content'];
-			}
+		// Max tokens are limited later to max tokens specified in the admin settings so here we just request PHP_INT_MAX
+		try {
+			$completion = $this->openAiAPIService->createChatCompletion($this->userId, $prompt, 1, $adminModel, PHP_INT_MAX, false);
+		} catch (Exception $e) {
+			throw new Exception('OpenAI/LocalAI request failed: ' . $e->getMessage());
```

Comment:
The TextProcessing API expects a RuntimeException I believe

Reply:
Thanks, it's fixed!